test/IntegrationTests/Cache/HybridCacheSetAndRemoveTests.cs

```diff
@@ -50,19 +53,34 @@ public async Task HybridCacheSerializesDateTime()
     {
         // Arrange
         var key = MethodKey();
-        var now = DateTime.UtcNow;
+        const string invariant = "2025-05-15T17:18:58.7503097Z";
+        var date = DateTime.Parse(invariant, CultureInfo.InvariantCulture);
 
         var options = new HybridCacheEntryOptions
         {
             Expiration = TimeSpan.FromMinutes(10)
         };
 
         // Act - Store the complex object in the cache
-        await HybridCache.SetAsync(key, now, options);
+        await HybridCache.SetAsync(key, date, options);
 
-        // Assert - Retrieve the object and verify it matches
+        // Assert - date is serialized as expected
+        var writer = new ArrayBufferWriter<byte>();
+        new NatsUtf8PrimitivesSerializer<DateTime>().Serialize(writer, date);
+        var serializedBytes = writer.WrittenSpan.ToArray();
+
+        var kvStore = await NatsConnection.CreateKeyValueStoreContext().GetStoreAsync("cache");
+        NatsJsonContextSerializer<CacheEntry> cacheEntrySerializer = new(CacheEntryJsonContext.Default);
+        var kvEntry = await kvStore.GetEntryAsync(key, serializer: cacheEntrySerializer);
+        Assert.NotNull(kvEntry.Value?.Data);
+
+        // HybridCache adds additional data to the front of the serialized value, so we're matching only the relevant data
+        var length = serializedBytes.Length;
+        Assert.Equal(serializedBytes[^length], kvEntry.Value.Data[^length]);
```

Comment:
nit- use Equivalent for collections (I guess Equal worked here but normally for collections need Equivalent) and also no need for ^length on serializedBytes

```
Assert.Equivalent(serializedBytes, kvEntry.Value.Data[^serializedBytes.Length], strict: true);
```